### PR TITLE
Construct canonical word-cube combinatorial-line hypergraphs

### LINKED ISSUE #2848: [Combinatorics on words] Construct finite word-cube combinatorial-line hypergraphs
## Parent gap / RFC

A source-grounded gap from solved Erdős #1090 and the finite Hales-Jewett vocabulary. The existing finite-abelian 3-AP hypergraph proposal (#2827) exactly covers ternary combinatorial lines after identifying `[3]` with `Z/3Z`; it does not construct the arbitrary-alphabet `k`-point lines used by Hales-Jewett arguments. Word/substitution operations likewise do not provide the complete finite cube or its line incidence.

## Public mathematical postcondition

Given integers `q >= 2` and `d >= 1`, return the complete canonical q-uniform hypergraph of combinatorial lines in the word cube `[q]^d`.

## Proposed operation ID

`words.combinatorial_line_hypergraph.compute`

## Input

An alphabet cardinality `q` and word dimension `d`. The canonical alphabet is the ordered digits `0,...,q-1`; an explicit alphabet relabelling belongs to a caller conversion, not a second mathematical cube.

## Result

A source-bound canonical value with every length-d word as a labelled vertex and one q-vertex edge for every pattern whose coordinates are either fixed digits or a common wildcard, with at least one wildcard. The edge order follows canonical pattern order and the vertices within an edge follow wildcard value `0,...,q-1`.

## Completeness semantics

Every admitted result includes exactly all cube vertices and all nonconstant combinatorial lines. The exact counts are `q^d` vertices and `(q+1)^d-q^d` line patterns/edges. Requests outside a preflighted expansion envelope reject; no partial hypergraph, timeout, or omitted line can establish line-freeness.

## Bounds and cost class

The semantic domain is every finite q-ary word cube. Admission must separately bound `q^d` vertex count, `((q+1)^d-q^d)` edge count, q times that many incidence slots, word-coordinate storage `d*q^d`, pattern-generation work, canonical labels, and serialized result bytes. Use checked exponentiation before allocation.

## Execution ownership

- Request-bounds owner and execution-plan quantities: combinatorics-on-words domain; vertex, edge, incidence, word-coordinate, and byte counts.
- Maintained backend or Jacobian kernel adapter: bounded Jacobian-owned product/pattern enumeration kernel.
- Canonical result construction and malformed-backend handling: domain-owned canonical digit-word labels, wildcard-pattern ordering, and q-edge rows.
- Independent-result verifier: none; the hypergraph is generated directly from q and d.
- Serialized-result bound and transport projection: all rows are preflighted before enumeration; MCP projects the same value.
- Native/MCP semantic parity and transport-only differences: identical exact cube and line semantics.

## Acceleration and exact representation

Generate each wildcard pattern once and derive its q vertices directly, rather than testing q-subsets of the cube. A compact internal base-q encoding may be used, but public vertices remain canonical word coordinates so later colouring and geometric-projection callers retain the source axis.

## Maintained implementation

A bounded Jacobian-owned enumeration kernel is proportionate: this is a total explicit finite construction with a closed-form preflight count, not an optimization or theorem solver.

## Defining invariant

Every returned edge has q distinct word vertices. For each edge there is exactly one nonempty wildcard mask and one fixed-digit assignment such that all vertices agree at fixed coordinates and take every common alphabet value at wildcard coordinates. Conversely every such pattern produces exactly one returned edge.

## Evidence plan

Test `q=2,d=1`, `q=3,d=1`, `q=3,d=2`, nontrivial `q=4` examples, count identities, coordinate/pattern reconstruction, distinct vertex and edge rows, and exact-boundary rejection. Cross-check small cases by independently enumerating all wildcard patterns and by verifying that the q=3 edge family agrees with the restricted coordinate-direction combinatorial-line subfamily, without claiming equality to all finite-group 3-APs.

## Why this is not caller-side composition

A caller can construct one chosen line but cannot obtain the complete cube, source-bound line family, and canonical incidence required for line-free experiments or finite colourings without recreating the same exponential construction. #2827 has a different input carrier and a different complete relation: all finite-abelian 3-APs, not arbitrary-alphabet word-cube lines.

## Deferred operation family

Do not add a Hales-Jewett decision/theorem solver, density-Hales-Jewett solver, maximum line-free-set optimizer, edge-colouring Ramsey solver, generic projection-to-the-plane routine, geometric realizability solver, arbitrary word-pattern DSL, or persistent cube search. These are caller composition, separate optimization, or theorem layers.

## Source motivation

Solved Erdős #1090 follows by applying Hales-Jewett to `[k]^n` and then taking a generic plane projection. The reusable finite ingredient is the combinatorial-line hypergraph before either the global Ramsey theorem or the geometry construction.

- https://www.erdosproblems.com/1090
- https://www.erdosproblems.com/171
- https://github.com/morluto/jacobian/issues/2827

### PR DESCRIPTION
**Pull request by morluto**
## Summary

Implements `words.combinatorial_line_hypergraph.compute` for #2848.

The operation constructs every standard Hales--Jewett combinatorial line in `[q]^d`: one nonempty wildcard-coordinate set, fixed values on the remaining coordinates, and the `q` words obtained by assigning one common alphabet value to every wildcard coordinate.

## Design

The result retains both representations callers need:

- canonical coordinate words and explicit line provenance (`wildcard_positions`, `fixed_coordinates`, and line vertices);
- a `FiniteHypergraph` projection with injective coordinate labels and stable edge IDs.

Explicit provenance makes the construction source-bound and distinguishes the standard relation from geometric or affine line families. In particular, `[3]^2` has six one-coordinate lines and one all-wildcard diagonal, hence 7 lines. This follows the wildcard definition in the density Hales--Jewett literature and the issue postcondition `(q+1)^d - q^d`.

Request admission derives vertex, edge, and incidence counts before enumeration and applies the actual `FiniteHypergraph` carrier limits. The native and MCP paths share that admission. This admits useful boundary cases such as `[2]^8` (256 vertices and 6,305 lines) while rejecting an unrepresentable request before kernel work.

## Validation

- `make handoff-scoped LANE=math TESTS=tests/math/combinatorics/word_cubes/test_combinatorial_line_hypergraph.py PATHS="src/jacobian/math/combinatorics/word_cubes tests/math/combinatorics/word_cubes/test_combinatorial_line_hypergraph.py"` (16 passed)
- `make test-catalog TESTS=tests/catalog` (113 passed)
- defining-invariant coverage reconstructs every line from its unique wildcard pattern and checks the generic hypergraph projection
- accepted carrier-boundary and native rejection cases cover request admission

Closes #2848

### CODE REVIEW

File: src/jacobian/math/combinatorics/word_cubes/operations.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    CombinatorialLineHypergraphResult,
+)
+
+__all__ = ["construct_combinatorial_line_hypergraph"]
+
+
+def construct_combinatorial_line_hypergraph(
+    alphabet_size: int,
+    dimension: int,
+) -> CombinatorialLineHypergraphResult:
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/combinatorics/word_cubes/operations.py:27]: Fixed on the PR branch: the exported native constructor now calls the shared owner admission before word and line enumeration, matching MCP constraints.
